Source/ShoppingLists/Mixins/ScrollList.lua

```diff
@@ -14,24 +14,64 @@ function AuctionatorScrollListMixin:OnLoad()
   })
 
   self:Register(self, { Auctionator.ShoppingLists.Events.ListItemDeleted })
+
+  self:InitSearch(
+    function(results)
+      self:EndSearch(results)
+    end,
+    function(current, total)
+      self.ResultsText:SetText("Searching for items in " .. self.currentList.name .. " (" .. current .. "/" .. total .. ")")
+    end
+  )
+end
+
+function AuctionatorScrollListMixin:OnEvent(eventName, ...)
+  self:OnSearchEvent(eventName, ...)
 end
 
 function AuctionatorScrollListMixin:EventUpdate(eventName, eventData)
   if eventName == Auctionator.ShoppingLists.Events.ListSelected then
     self.currentList = eventData
 
+    self:StartSearch()
+
     -- Propogate events to children
     self:Fire(eventName, eventData)
 
     self:RefreshScrollFrame()
   elseif eventName == Auctionator.ShoppingLists.Events.ListItemAdded then
     self:RefreshScrollFrame()
   elseif eventName == Auctionator.ShoppingLists.Events.ListItemDeleted then
-    Auctionator.Debug.Message("I got the deleted message")
     self:RefreshScrollFrame()
   end
 end
 
+function AuctionatorScrollListMixin:StartSearch()
+  self.ResultsText:SetText("Searching for items in " .. self.currentList.name .. "...")
+  self.ResultsText:Show()
+
+  self.SpinnerAnim:Play()
+  self.LoadingSpinner:Show()
+
+  local searchTerms = {}
+
+  for _, name in ipairs(self.currentList.items) do
+    table.insert(searchTerms, name)
+  end
+
+  self:Search(searchTerms)
+end
+
+function AuctionatorScrollListMixin:EndSearch(results)
+  C_AuctionHouse.SearchForItemKeys(results, {sortOrder = 1, reverseSort = false})
+
+  -- Lame but I can't figure out how to tell when we get all the results in the main frame?
```

Comment:
Waiting for a AUCTION_HOUSE_BROWSE_RESULTS_UPDATED should do it

Reply:
Yeah but its firing for the search as well... I guess I could count the events?

Reply:
At this point all the search events have fired, so the next one is for this search

Reply:
Pushed a commit showing the intent